File: Linked List/0024.Swap Nodes in Pairs/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        Deque<ListNode> stk = new ArrayDeque<>();
        ListNode temp = new ListNode();
        ListNode cur = head;
        head = temp;
        while (cur != null && cur.next != null) {
            stk.addFirst(cur);
            stk.addFirst(cur.next);
            cur = cur.next.next;
            temp.next = stk.pop();
            temp = temp.next;
            temp.next = stk.pop();
            temp = temp.next;
        }
        temp.next = cur;
        return head.next;
    }
}

File: Linked List/0024.Swap Nodes in Pairs/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        stk = []
        temp, cur = ListNode(), head
        head = temp
        while cur and cur.next:
            stk.append(cur)
            stk.append(cur.next)
            cur = cur.next.next
            temp.next = stk.pop()
            temp = temp.next
            temp.next = stk.pop()
            temp = temp.next
        temp.next = cur
        return head.next